src/pages/workspaces/workspace/useDeleteWorkspaceState.ts

```diff
@@ -35,83 +53,153 @@ export interface DeleteWorkspaceHookArgs {
+        const [{ acl }, { usageInBytes }] = await Promise.all([
           Ajax(signal).Workspaces.workspace(workspaceInfo.namespace, workspaceInfo.name).getAcl(),
           Ajax(signal).Workspaces.workspace(workspaceInfo.namespace, workspaceInfo.name).bucketUsage()
         ])
-        setApps(currentWorkspaceAppList)
         setCollaboratorEmails(_.without([getUser().email], _.keys(acl)))
         setWorkspaceBucketUsageInBytes(usageInBytes)
-      } else {
-        const currentWorkspaceAppList = await Ajax(signal).Apps.listAppsV2(workspaceInfo.workspaceId)
-
-        // temporary hack to prevent orphaning resources on Azure:
-        // change each app to status: 'disallow' which will cause this modal to think they are undeletable
-        const hackedAppList = _.map(_.set('status', 'disallow'), currentWorkspaceAppList) as DeleteWorkspaceModalLeoApp[]
-        setApps(hackedAppList)
-
-        // Also temporarily disable delete if there are any controlled resources besides the expected workspace storage container.
-        const controlledResources = await Ajax(signal).WorkspaceManagerResources.controlledResources(workspaceInfo.workspaceId)
-        setControlledResourcesExist(controlledResources.resources.length > 1)
       }
     })
     load()
+
+    clearTimeout(checkAzureResourcesTimeout.current)
   })
 
   const hasApps = () => {
-    return deletableApps !== undefined && nonDeletableApps !== undefined &&
-        (!_.isEmpty(deletableApps) ||
-            !_.isEmpty(nonDeletableApps))
+    return (workspaceResources !== null && !_.isEmpty(workspaceResources.apps))
+  }
+
+  const hasRuntimes = () => {
+    return (workspaceResources !== null && !_.isEmpty(workspaceResources.runtimes))
   }
 
-  const isDeleteDisabledFromResources = (hasApps() && !_.isEmpty(nonDeletableApps)) || controlledResourcesExist
+  const isDeleteDisabledFromResources = workspaceResources !== null && (
+    (hasApps() && !_.isEmpty(workspaceResources.nonDeleteableApps)) || (hasRuntimes() && !_.isEmpty(workspaceResources.nonDeleteableRuntimes))
+  )
 
   const deleteWorkspace = async () => {
+    if (isDeleteDisabledFromResources) {
+      throw new Error('Workspace contains non-deletable resources')
+    }
+
     try {
       setDeleting(true)
-      if (isGoogleWorkspace(hookArgs.workspace)) {
+      if (isGoogleWorkspace(hookArgs.workspace) && workspaceResources) {
         await Promise.all(
-          _.map(async app => await Ajax().Apps.app(app.cloudContext.cloudResource, app.appName).delete(), deletableApps)
+          _.map(async app => await Ajax().Apps.app(app.cloudContext.cloudResource, app.appName).delete(), workspaceResources.deleteableApps)
         )
       }
+
+
       await Ajax().Workspaces.workspace(workspaceInfo.namespace, workspaceInfo.name).delete()
       hookArgs.onDismiss()
       hookArgs.onSuccess()
     } catch (error) {
-      reportError('Error deleting workspace', error)
       setDeleting(false)
+      throw error
```

Comment:
I am a bit fuzzy on the correct mechanism for error handling in hooks such as this. I think throwing an exception and letting the UI manifest it as an error notification is the right call vs a direct call to `reportError` is the right approach, but would love a second opinion.